docs/src/main/asciidoc/getting-started-dev-services.adoc

```diff
@@ -0,0 +1,326 @@
+- If using cmd , (don't use backward slash `\` and put everything on the same line)
+- If using Powershell , wrap `-D` parameters in double quotes e.g. `"-DprojectArtifactId=getting-started"`
+
+For an explanation of what's in the generated application, see the xref:getting-started.adoc[First application guide].
+
+=== Running the application
+
+Launch the application in dev mode
+----
+mvn quarkus:dev
+----
+
+Once the application is up, visit http://localhost:8080/hello. It should show a "Hello from RestEasy Reactive" message.
+
+=== Accepting user input
+
+Let's make the application a bit more interactive.
+Open the project in your IDE and navigate to `src/main/java/org/acme/GreetingResource.java'
+Add a query param in the `hello` method.
+
+----
+public String hello(@QueryParam("name") String name) {
+return "Hello " + name;
+}
+----
+
+Visit http://localhost:8080/hello?name=Bloom.
+
+You should see a personalised message: `Hello Bloom`.
+
+=== Fixing the tests
+
+In your Quarkus terminal, type 'r' to run the tests. You should see
+that your application changes broke the tests!
+
+To fix the tests, open `src/test/java/org/acme/GreetingResource.test`
+and replace
+
+----
+             .body(is("Hello from RESTEasy Reactive"));
+----
+
+with
+
+----
+             .body(containsString("Hello"));
+----
+
+This still validates the rest endpoint, but it's more flexible
+about the expected output.
+You should see in your terminal that the tests are now passing.
+
+== Adding persistence
+
+=== Creating a Panache Entity
+
+1. To add the persistence libraries, run
+----
+quarkus ext add hibernate-orm-panache jdbc-postgresql
+----
+
+The application will record the names of people it greets. Define a simple Hibernate Entity
+by creating a `Greeting.java` class. Add the following content:
+
+----
+@Entity
+public class Greeting extends PanacheEntity {
+String name;
+}
+----
+
+The entity makes use of xref:{doc-guides}hibernate-orm-panache.adoc[Panache], a layer on top of Hibernate.
+ Extending `PanacheEntity` brings in a range of methods for reading, writing, and finding data.
+ Because all the data access methods are on the `Greeting` entity, rather than on a separate data access class,
+ this is an example of the active record pattern.
+
+The `Greeting` table will have one column, a field called `name`.
+
+=== Writing data
+
+To use the new entity, update the `hello` method to start writing some data.
+
+Change the method to the following:
+
+----
+@GET
+@Transactional
+@Produces(MediaType.TEXT_PLAIN)
+public String hello(@QueryParam("name") String name) {
+   Greeting greeting = new Greeting();
+   greeting.name = name;
+   greeting.persist();
+   return "Hello " + name;
+}
+----
+
+Don't forget the `@Transactional` method, which ensures writes are wrapped
+in a transaction.
+
+Generally, you shouldn't do state updates on a `GET` REST method, but here it makes
+trying things out simpler. Let's assume what's being written is an logging "side effect",
```

Comment:
Ah ah... ok.... 

Does adding another @POST method adds so much complexity?

Reply:
Ha! It looks like we *both* should have read down before typing. :)

The Java code for POST and a second method isn't so complex, but I didn't want to have to talk through doing things on the command line with CURL. And, it still feels like removing the ability for people to get a greeting in a browser is a functional regression - because of our changes, our greeting app has lost its ability to issue greetings. Instead of being a greeting-giver, it's now a greeting-receiver.